packages/selfcare-client-users-updater-consumer/src/services/selfcareClientUsersUpdaterProcessor.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  getInteropHeaders,
+  logger,
+  RefreshableInteropToken,
+  userRole,
+} from "pagopa-interop-commons";
+import { EachMessagePayload } from "kafkajs";
+import {
+  generateId,
+  CorrelationId,
+  genericInternalError,
+} from "pagopa-interop-models";
+import { AuthorizationProcessClient } from "../clients/authorizationProcessClient.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function selfcareClientUsersUpdaterProcessorBuilder(
+  refreshableToken: RefreshableInteropToken,
+  authorizationProcessClient: AuthorizationProcessClient,
+  productId: string,
+  allowedOriginsUuid: string[]
+) {
+  return {
+    async processMessage({
+      message,
+      partition,
+    }: EachMessagePayload): Promise<void> {
+      const correlationId: CorrelationId = generateId();
+
+      const loggerInstance = logger({
+        serviceName: "selfcare-client-users-updater-consumer",
+        correlationId,
+      });
+
+      try {
+        loggerInstance.info(
+          `Consuming message for partition ${partition} with offset ${message.offset}`
+        );
+
+        if (!message.value) {
+          loggerInstance.warn(
+            `Empty message for partition ${partition} with offset ${message.offset}`
+          );
+          return;
+        }
+
+        const stringPayload = message.value.toString();
+        const jsonPayload = JSON.parse(stringPayload);
+
+        // Process only messages of our product
```

Comment:
I don't know if this comment line have a great value, the follow code is clear enough, anyway  in this case we can change it with a message with example 
```suggestion
        // Process only messages of our productId, eg : ....
```

Reply:
Yes for me this line can be deleted too